yo/app/scripts/directives/copy-or-migrate-mappings.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (C) 2013 – 2015  SLUB Dresden & Avantgarde Labs GmbH (<code@dswarm.org>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+angular.module('dmpApp')
+    .controller('CopyOrMigrateMappingsCtrl', function($scope, $window, $http, $q, $modal, $modalInstance, loDash, ngProgress, showAlert, DataModelResource, ProjectResource, dataModels, projects) {
+
+        $scope.internalName = 'Copy Or Migrate Mappings Widget';
+
+        $scope.alerts = [];
+
+        $scope.closeAlert = function(idx) {
+            if (angular.isDefined(idx) && $scope.alerts.length > idx) {
+                $scope.alerts.splice(idx, 1);
+            }
+        };
+
+        $scope.dataModels = dataModels;
+        $scope.projects = projects;
+
+        $scope.selectedDataModel = [];
+        $scope.selectedProject = [];
+
+        var isNewProjectsHasBeenCreated = false;
+
+        $scope.updateGridData = function() {
+
+            DataModelResource.query({ format : 'medium' }, function(results) {
+
+                $scope.dataModels = loDash.filter(results, 'data_resource');
+
+            }, function() {
+                $scope.dataModels = '';
```

Comment:
are those strings? I think they're actually arrays; JS might be dynamic, but we should just change the types of variables.